src/app/business/block/cloud-block-service/modify/cloud-volume-modify.component.ts

```diff
@@ -0,0 +1,297 @@
+import { Router,ActivatedRoute } from '@angular/router';
+import { Component, OnInit, ViewContainerRef, ViewChild, Directive, ElementRef, HostBinding, HostListener, Input } from '@angular/core';
+import { I18NService, Consts, MsgBoxService, HttpService, Utils } from 'app/shared/api';
+import { ConfirmationService, ConfirmDialogModule, MenuItem, Message} from '../../../../components/common/api';
+import { FormGroup, Validators, FormBuilder, FormControl, FormArray } from '@angular/forms';
+import { trigger, state, style, transition, animate} from '@angular/animations';
+import { I18nPluralPipe } from '@angular/common';
+import { CloudBlockServiceService} from '../cloud-block-service.service';
+import { BucketService } from '../../buckets.service';
+
+
+let _ = require("underscore");
+@Component({
+    selector: 'app-cloud-volume-modify',
+    templateUrl: './cloud-volume-modify.component.html',
+    styleUrls: [],
+    providers: [ConfirmationService, MsgBoxService ]
+})
+export class CloudVolumeModifyComponent implements OnInit{
+    allCloudVolumes: any = [];
+    modifyVolumeForm: FormGroup;
+    allBackends: any = [];
+    backendsOption = [];
+    allTypes = [];
+    selectType;
+    listedBackends: any;
+    allAWSVolumes: any = [];
+    selectedVolumes: any = [];
+    selectedBackends = [];
+    selectedVolume: any;
+    selectedVolumeId: any;
+    backendId: any;
+    volumeTypeOptions: any = [];
+    selectedVolType;
+    msgs: Message[];
+    menuItems: MenuItem[];
+    menuDeleDisableItems : MenuItem[];
+    createVolumeDisplay: boolean = false;
+    enableEncryption: boolean = false;
+    isVisible: boolean = false;
+    label = {
+        name: "Name",
+        description: "Description",
+        size: "Size",
+        type: "Type",
+        encrypted: "Enable Encryption",
+        encryptionSettings: "Encryption Settings",
+        state: "State",
+        volumeId: "Volume ID",
+        availabilityZone: "Availability Zone",
+        createdAt: "Created At",
+        iops: "IOPS",
+        tags: "Tags",
+    };
+    errorMessage = {
+        "name": { 
+            required: "Name is required",
+            minlength: "The volume name should have minimum 3 characters.",
+            maxlength: "The volume name should have maximum 64 characters.",
+            pattern: "Must start with a character. Can contain alphabets, numbers and underscore. No special characters allowed."
```

Comment:
These constraints are taken from AWS?

Reply:
I believe so although I am not sure. 
In this file these constraints are not required. Name cannot be modified so the form does not need validation. I will remove this.

Reply:
ok